feat(createLambdaExtensionClient): Allow metrics propagation via Datadog Lambda Extension (#196)
seek-oss/datadog-custom-metrics: README.md

> ## Table of contents

Reviewer: Should we remove the ToC given GitHub generates one for us now?

<img width="341" alt="Screenshot 2023-03-18 at 7 18 20 pm" src="https://user-images.githubusercontent.com/25572311/226094092-76ed12a3-e4ad-4e2d-8a0f-1276aecdc760.png">

const metricsClient = createStatsDClient(StatsD, config, errorHandler);
```

### `createLambdaExtensionClient`

> `createLambdaExtensionClient` creates a [lambda extension](https://docs.datadoghq.com/serverless/libraries_integrations/extension/) client.

Reviewer: ```suggestion
`createLambdaExtensionClient` creates a [Lambda extension](https://docs.datadoghq.com/serverless/libraries_integrations/extension/) client.
```

> This is intended for AWS Lambda functions and is a replacement for `createCloudWatchClient`.

Reviewer: Can we update the deprecation notice on `createCloudWatchClient` to advise users to switch to `createLambdaExtensionClient`?

> This client will only submit metrics as a [distribution](https://docs.datadoghq.com/metrics/distributions/) which enables broader aggregations for percentiles (p50, p75, p90 etc).

Reviewer: Would being more specific here be helpful?

```suggestion
This client will only submit metrics as a [distribution](https://docs.datadoghq.com/metrics/distributions/) which enables globally-accurate aggregations for percentiles (p50, p75, p90 etc).
```

```typescript
import { createLambdaExtensionClient } from 'seek-datadog-custom-metrics';

// Expects `name` and `metrics` properties
import config from '../config';

// Returns a standard hot-shots StatsD instance
const { metricsClient, withLambdaExtension } =
  createLambdaExtensionClient(config);

> export const handler = withLambdaExtension(
>   (event: unknown, lambdaContext: LambdaContext) => {

Reviewer: ```suggestion
export const handler = withLambdaExtension(
  (event, _ctx) => {
```
